src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/Utilities/DeviceTest.cs

```diff
@@ -43,12 +43,30 @@ protected override void CleanupTest ()
 				}
 			}
 
+			UnInstallTestApp ();
+
 			ClearAdbLogcat ();
 
 
 			base.CleanupTest ();
 		}
 
+		protected static void UnInstallTestApp ()
+		{
+			if (TestPackageName.TryGetValue (TestContext.CurrentContext.Test.ID, out string packageName))
+				RunAdbCommand ($"uninstall {packageName}");
+		}
+
+		protected static string GeneratePackageName ()
+		{
+			if (TestPackageName.TryGetValue (TestContext.CurrentContext.Test.ID, out string packageName))
+				return packageName;
+
+			packageName = $"com.xamarin.test{TestContext.CurrentContext.Test.ID.Replace ("-", "")}";
```

Comment:
What is the value of `Test.ID` compared to `Test.Name`? Is it a GUID?

Reply:
We end up with id's like `com.xamarin.test01022`. My main concern with using the `Test.Name` is it includes all the arguments and even with replacing invalid chars we'd end up with something like 
```
com.xamarin.ApplicationRunsWithDebuggerAndBreaksTrueAssembliesFalsenull
```
which I suppose isn't that bad.